article/_posts/2020-11-19-AlphaMuGoZero.md

```diff
@@ -0,0 +1,181 @@
+
+Above is the ELO rating for AlphaGO and European Go champion Fan Hui, as well as other Go programs and different versions of AlphaGo. AlphaGo went on to [beat Lee Sedol 4-1 afterward](https://www.youtube.com/watch?v=WXuK6gekU1Y).
+
+## AlphaGo Zero
+
+In a subsequent paper, Deepmind presented AlphaGo Zero, which achieves higher performance than AlphaGo using fewer priors.
+
+![](/article/images/alpha/fig4.jpg)
+> Silver, D., Schrittwieser, J., Simonyan, K. et al. Mastering the game of Go without human knowledge. Nature 550, 354–359 (2017). https://doi.org/10.1038/nature24270
+
+A couple of differences are to note:
+- No supervised learning is used to initialize policies
+- A single policy is used
+- A single network is used for both the policy and the value function
+- The network is fed raw board configurations instead of handcrafted features
+
+This new version of the algorithm takes the form
+
+$$p, v = f_\theta(s)$$
+
+, meaning that the network outputs both action probabilities $$p$$ and value estimation $$v$$ for states $$s$$. During training, at each move, the network performs MCTS simulations following
+
+$$a_t = \argmax_a U(s,a)$$
+
+with
+
+$$U(s,a) = Q(s,a) + c \cdot p(s,a) \cdot \frac{\sqrt{\sum_b N(s,b)}}{1+N(s,a)}$$
+
+$$Q(s,a) = \frac{1}{N(s,a)}\sum_{s'|s,a\rightarrow s'}v(s')\; \text{(an average of the values of the nodes from s' to s)}$$ 
+
+and $$c$$ a parameter promoting exploration. However, instead of performing rollouts, the network uses $$v(s_L)$$ to predict the outcome. MCTS' execution gives the tuple $$(s, \pi, p, \_)$$ where $$s$$ is the game's state, $$\pi$$ corresponds to the normalized state/actions visitations
+
+$$\pi = \frac{N(s, \cdot)^{1/\tau}}{\sum_b N(s,b)^{1/\tau}}$$
+
+with $$\tau$$ a temperature parameter controlling exploration. $$p$$ is the action probabilities for state $$s$$, and $$\_$$ is later filled with $$z \in \pm 1$$ depending on the game's outcome. Then, the loss function
+
+$$l = (z - v)^2 - \pi^T \log p + c \|\theta\|^2$$
+
+where $$c$$ in this case is a regularization term, is used to train the network. We can interpret the loss as cross-entropy forcing the policy to match MCTS' behavior.
+
+![](/article/images/alpha/fig5.jpg)
+> Silver, D., Schrittwieser, J., Simonyan, K. et al. Mastering the game of Go without human knowledge. Nature 550, 354–359 (2017). https://doi.org/10.1038/nature24270
+
+From the above figure, we can see that the algorithm can surpass AlphaGo after only ~40h of training. Interestingly enough, we can see that AlphaGo Zero is not really good at prediction moves, but quite good at predicting game outcome. A supervised learning version of the algorithm is shown for comparison. Because it manages to outperform AlphaGo (and therefore the best human player is ELO is transitive), the results seem to imply that AlphaGo Zero learns to play differently than human players (which could explain moves 37 and 78). 
```

Comment:
(...) at move prediction
or
(...) at predicting moves
?

Reply:
Fixed in 137ef52